packages/headless/src/controllers/knowledge/generated-answer/headless-answerapi-generated-answer.ts

```diff
@@ -113,6 +159,18 @@ export function buildAnswerApiGeneratedAnswer(
     reset() {
       engine.dispatch(resetAnswer());
     },
+    async sendFeedback(feedback) {
+      if (typeof feedback !== 'object') {
+        return;
+      }
```

Comment:
This is a Type Narrowing

Reply:
Maybe add a TODO to remove it as part of SFINT-5585

Reply:
Or, another option would be to define the `AnswerApiGeneratedAnwser` interface (line 33 of this file) as:

```typescript
export interface AnswerApiGeneratedAnswer extends Omit<GeneratedAnswer, 'sendFeedback'> {
  /**
   * Resets the last answer.
   */
  reset(): void;
  /**
   * Sends feedback about why the generated answer was not relevant.
   * @param feedback - The feedback that the end user wishes to send.
   */
  sendFeedback(feedback: GeneratedAnswerFeedbackV2): void;
}
```

then no type narrowing would be needed

Reply:
I defined the return type of the controller as you suggested :+1: